File: match-backend/src/schema/Project.ts
import mongoose, { Schema, Document } from 'mongoose';

// Define the Project interface
interface IProject extends Document {
    name: string;
    description: string;
    owner: mongoose.Types.ObjectId; // Reference to a User/Student
    desiredSkills: string[];
    languages: string[];
    field: string;
    location?: string; // Optional
}

// Create the Project Schema
const projectSchema = new Schema<IProject>({
    name: { type: String, required: true },
    description: { type: String, required: true },
    owner: { type: Schema.Types.ObjectId, ref: 'Student', required: true }, // Reference to the Student model
    desiredSkills: { type: [String], default: [] },
    languages: { type: [String], default: [] },
    field: { type: String, required: true },
    location: { type: String, default: null } // Optional
});

// Create the Project Model
const Project = mongoose.model<IProject>('Project', projectSchema);

export default Project;

File: match-recommender/src/models/project.py
from typing import List, Optional
from bson import ObjectId

# Project schema definition
project_schema = {
    "$jsonSchema": {
        "bsonType": "object",
        "required": ["name", "description", "field"],
        "properties": {
            "name": {
                "bsonType": "string",
                "description": "Name of the project - required"
            },
            "description": {
                "bsonType": "string", 
                "description": "Project description - required"
            },
            "members": {
                "bsonType": "array",
                "description": "List of project members",
                "items": {
                    "bsonType": "object",
                    "required": ["user", "role"],
                    "properties": {
                        "user": {
                            "bsonType": "objectId",
                            "description": "Reference to User document"
                        },
                        "role": {
                            "bsonType": "string",
                            "description": "Role of the member"
                        }
                    }
                }
            },
            "desiredSkills": {
                "bsonType": "array",
                "description": "List of desired skills",
                "items": {
                    "bsonType": "string"
                }
            },
            "languages": {
                "bsonType": "array",
                "description": "List of programming languages", 
                "items": {
                    "bsonType": "string"
                }
            },
            "field": {
                "bsonType": "string",
                "description": "Project field - required"
            },
            "location": {
                "bsonType": ["string", "null"],
                "description": "Project location - optional"
            }
        }
    }
} 